types/ember__controller/test/main.ts

```diff
@@ -12,3 +12,17 @@ Controller.extend({
         this.toggleProperty('isExpanded');
     },
 });
+
+class CustomController extends Controller {
+    queryParams = ['category'];
```

Comment:
So this is *incredibly* annoying, but I was thinking about this and went to describe how to test it more effectively and… this change doesn't actually make `queryParams` be a `ReadonlyArray`, since subclasses can make it a regular array—and in fact that's what we're doing here. 😩 So we should close this and revert the change on Ember itself.

[playground](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEkoGdnwKIA8oFsAOECA3gLABQ88AjnsgFzwBKIUwA9gHYQCeAgjDCjcAPMgAuMAJYcA5gD5yAX3LlQSOPEJj4IbPkIMsuAiADc5XccIA6Gsmt4ArsgAWACgBELkBAhsPAJQqZEio8ABibGw6mGIgHMBoRvrE5JR28AC88ADaXj5+HgC65mTKZKrg0Bpa8ABmUQyRbKUNbLa0Ds7u+b7+AaZAA)

Reply:
Well all I really wanted was to be able to assign a frozen object, so it works for that, no?

Reply:
Ah, I misunderstood. Seems good.